OSVR-Unity/Assets/OSVRUnity/Sample/Scripts/SampleImaging.cs

```diff
@@ -0,0 +1,92 @@
+﻿/// OSVR-Unity Connection
+///
+/// http://sensics.com/osvr
+///
+/// <copyright>
+/// Copyright 2016 Sensics, Inc.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+/// </copyright>
+using System.Runtime.InteropServices;
+using UnityEngine;
+using System.Collections;
+using OSVR.ClientKit;
+using System;
+using Quaternion = UnityEngine.Quaternion;
+
+public class SampleImaging : OSVR.Unity.RequiresImagingInterface
+{
+    private int imageWidth = 640;
+    private int imageHeight = 480;
+
+    private OSVR.Unity.ImagingInterface imagingInterface;
+
+    private Texture2D videoTexture;
+
+    private Color32[] imageData;
+    private bool videoChanged;
+    private bool firstReport = false;
+
+    void Start()
+    {
+        firstReport = false;
+        imagingInterface = GetComponent<OSVR.Unity.ImagingInterface>();
+        imagingInterface.Interface.StateChanged += HandleChanged;
+
+    }
+
+    public void Update()
+    {
+        if (videoChanged)
+        {
+            videoTexture.SetPixels32(imageData);           
+            videoTexture.Apply();
+            videoChanged = false;
+        }     
+    }
+
+    private void HandleChanged(object sender, TimeValue timestamp, int sensor, ImagingState imageReport)
+    {
+        if(!firstReport)
+        {
+            firstReport = true;
+            initVideoTexture(imageReport.metadata);
+            return;
+        }
+        if (videoChanged)
```

Comment:
I saw this in the OSVR-Leap example and thought it was for performance reasons (since they have to do a lot more than just read the video frame), so I followed suit. I can't tell a difference, so I may take that out and remove the whole update function, handling everything in the callback.